### PR TITLE
Add NNG interface

### PR DESCRIPTION
**Pull request by EyeOfPython**
This allows external indexers to be built efficiently using an easy to use NNG interface

### CODE REVIEW

File: src/validation.cpp [hunk 1]
```diff
@@ -739,7 +739,14 @@ bool MemPoolAccept::AcceptSingleTransaction(const CTransactionRef &ptx,
         return false;
     }
 
-    GetMainSignals().TransactionAddedToMempool(ptx);
+    std::vector<Coin> spent_coins;
+    spent_coins.reserve(ptx->vin.size());
```
Comment (schancel) [inline src/validation.cpp:743]: Why do you need `spent_coins` here? Avoid having to look them up via more calls?

File: test/functional/interface_nng.py [hunk 2]
```diff
@@ -0,0 +1,688 @@
+#!/usr/bin/env python3
```
Comment (schancel) [inline test/functional/interface_nng.py:1]: I think the naming of this will trigger a testing error? There's some prefix checks.

**Comment by schancel** [comment]
There is a `depends` folder as well as the gitian descriptors that are required. I *believe* that based on what you said that `flatbuffers` would need to have a makefile in `depends/packages` and the gitian builds need `libnng-dev`

File: src/validation.cpp (hunk 1 above)
Comment (EyeOfPython) [reply src/validation.cpp:743]: Yes, spent coins are not indexed. The data is right there in the node, so this simplifies the indexer.

**Review by schancel — APPROVED** [review]

File: test/functional/interface_nng.py (hunk 2 above)
Comment (EyeOfPython) [reply test/functional/interface_nng.py:1]: No, this obeys the naming rules.

**Pull request merged**